tools/tasks.py

```diff
@@ -217,10 +216,13 @@ def __init__(self, description, callable, timeout=None, **kwargs):
 
     def execute(self, fp):
         """Run the task"""
-        print("log_file: {0}. ".format(self.log_file))
+        print(("log_file: {0}. ".format(self.log_file)))
         try:
             result = self.callable()
-            fp.write(result)
+            try:
+                fp.write(result.encode())
+            except (UnicodeEncodeError, AttributeError):
+                fp.write(result)
```

Comment:
If result.encode() has an error, what is fp.write(result) going to do?

Reply:
Its going to send the result as is. This is due to the fact that sometimes result is a string and sometimes bytes. This handles this case where it'll try to convert to bytes, however, if its already in byte format then it'll fall into the except case.

Reply:
It needs to be in bytes to send it